io/include/pcl/io/point_cloud_image_extractors.h

```diff
@@ -403,8 +403,8 @@ namespace pcl
       using ScalingMethod = typename PointCloudImageExtractorWithScaling<PointT>::ScalingMethod;
 
       public:
-        using Ptr = shared_ptr<PointCloudImageExtractorFromIntensityField<PointT> >;
-        using ConstPtr = shared_ptr<const PointCloudImageExtractorFromIntensityField<PointT> >;
+        using Ptr = std::shared_ptr<PointCloudImageExtractorFromIntensityField<PointT> >;
```

Comment:
This PR has inconsistent behavior with `shared_ptr`. I'd recommend to start a separate PR with all the `shared_ptr` changes, and going on ahead with the non-controversial ones ion this PR.

PS: For new PR, use `git reset` after creating a new branch from this PR, then selectively add the changes with `git add -p filename`. You can get the list of filenames by doing `git diff | grep 'shared_ptr'`. Or a similar workflow

Reply:
One difficulty I'm having is that many of the build errors only occur in the Ubuntu 18.04 Clang configuation. Is there a way to replicate that environment? (In CircleCI, I can `ssh` into a CI instance and experiment there, but apparently not in Azure.)

Reply:
This should be a better solution: For files that don't compile with the unqualified `shared_ptr` (which is what I assume you meant by 'non-controversial'), I placed a `using std::shared_ptr` at the top. Those in the `pcl` namespace should qualify themselves properly.

Reply:
You assumed right